src/components/layout/Analytics.astro

```
@@ -5,47 +5,187 @@
+      f.parentNode.insertBefore(j, f);
+    })(window, document, 'script', 'dataLayer', gtmId);
+  </script>
+)}
+
+{noConsent && hasGa && (
+  <>
+    <script is:inline async src={`https://www.googletagmanager.com/gtag/js?id=${gaId}`}></script>
+    <script is:inline define:vars={{ gaId }}>
+      window.dataLayer = window.dataLayer || [];
+      // eslint-disable-next-line prefer-rest-params
+      function gtag(){dataLayer.push(arguments);}
```

ESLint

[error] 'dataLayer' is not defined.

(no-undef)